tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -55,19 +54,22 @@ public void httpHandler(HttpServletRequest hsreq, HttpServletResponse hsres) {
 			
 			// 
 			if(request != null) {
+				WFSLogger.logInfo("WFS Server received GetCapabilities request.");
 				if(request.equals(WFSRequestType.GetCapabilities.toString())) {
 					getCapabilities(wr, hsreq);
 				}
 			}
 			
 			else{
-				wr.append("ncWFS on THREDDS - ");
-				wr.append(hsreq.getScheme() + "://" + hsreq.getServerName() + ":" + hsreq.getServerPort() + "/thredds/wfs");
+				WFSExceptionWriter owsExcept = new WFSExceptionWriter("WFS server error. REQUEST parameter is required.", "request", "MissingParameterValue", hsres);
+				owsExcept.write();
+				return;
 			}
 		}
 		
 		catch(IOException io) {
-			
+			WFSLogger.logError("ERROR: retrieval of writer failed.");
+			throw new RuntimeException("ERROR: retrieval of writer failed");

```

Comment:
If you would like to throw an exception here, it might help to add `io` as a parameter to the `RuntimeException` constructor. This could help something further up the stack gain insight into the cause of the problem, and thus perform more helpful recovery or error reporting.

Reply:
Ah yes, thanks for catching that Carl! Will pass it on through to the creation of Runtime Exception.